src/resources/pandoc/datadir/_json.lua

```diff
@@ -101,13 +116,13 @@ local function encode_table(val, stack)
     return "[" .. table.concat(res, ",") .. "]"
   elseif types["string"] then
     -- Treat as object
-    for k, v in pairs(val) do
+    for k, v in pairsByKeys(val) do
       table.insert(res, encode_string(k) .. ":" .. encode(v, stack))
     end
     stack[val] = nil
     return "{" .. table.concat(res, ",") .. "}"
   else
-    return "[]"
+    error( string.format("invalid table: unsupported key type %s", types[1]) )
```

Comment:
change in functionality to maintain compatibility with upstream tests, encoding `{ [ function() end ] = 12 }` in particular

Reply:
that change broke encoding empty tables which is not covered by upstream tests. I will fix that.

Reply:
fixed upstream as well and also added tests there :heavy_check_mark: